alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -286,23 +291,27 @@ class TransportIndexMonitorAction @Inject constructor(
                 return
             }
 
-            if (request.method == RestRequest.Method.PUT) return updateMonitor()
-
-            val query = QueryBuilders.boolQuery().filter(QueryBuilders.termQuery("${Monitor.MONITOR_TYPE}.type", Monitor.MONITOR_TYPE))
-            val searchSource = SearchSourceBuilder().query(query).timeout(requestTimeout)
-            val searchRequest = SearchRequest(SCHEDULED_JOBS_INDEX).source(searchSource)
-            client.search(
-                searchRequest,
-                object : ActionListener<SearchResponse> {
-                    override fun onResponse(searchResponse: SearchResponse) {
-                        onSearchResponse(searchResponse)
-                    }
+            if (request.method == RestRequest.Method.PUT) {
+                scope.launch {
+                    updateMonitor()
+                }
+            } else {
+                val query = QueryBuilders.boolQuery().filter(QueryBuilders.termQuery("${Monitor.MONITOR_TYPE}.type", Monitor.MONITOR_TYPE))
+                val searchSource = SearchSourceBuilder().query(query).timeout(requestTimeout)
+                val searchRequest = SearchRequest(SCHEDULED_JOBS_INDEX).source(searchSource)
+                client.search(
```

Comment:
same as one of the previous comment, get rid of callbacks if using suspend functions.

Reply:
I will refactor this class in a future PR to make it all aligned.